src/models/utils.ts

```diff
@@ -308,6 +312,13 @@ export function isSameConnectionInfo(
     conn: IConnectionInfo,
     expectedConn: IConnectionInfo,
 ): boolean {
+    // If connection info has an id, use that to compare connections
+    const connId = (<IConnectionProfile>conn).id;
```

Comment:
Nit: good practice to use `as` instead of `<>` for casting just due to JSX/TSX portability.